### ISSUE TITLE
Add config option for setting a custom base box & version

### ISSUE THREAD

**Issue by roborourke** [Chassis/Chassis#827]
There is a use-case that has come up regarding using Chassis with a non standard configuration and a lot of extensions. With [Altis Local Chassis](https://github.com/humanmade/altis-local-chassis) we do the following:

- override PHP version
- add many many many extensions

This all takes a while to provision because it is uninstalling PHP, and adding lots of extra software etc...

If we could use the basebox builder to pre-install all of these things then provisioning Chassis with our customisations would be much faster, albeit at the cost of a larger download.

Another reason that this would be helpful is when providing for developers working behind strict proxy / firewall rules that block certain URLs or file types.

The specific case is Elasticsearch plugins, these are zip files and are failing to download for one of our users preventing Chassis from successfully provisioning. If they were already bundled on the box image then we could circumvent the problem.

I wanted to get your thoughts @rmccue and @BronsonQuick. It's technically easy to add but is the use case something we want to support in Chassis core?

I think it may be useful to others to help find time gains by creating pre-configured base boxes for their typical dev setup.

**Comment by BronsonQuick** [comment]
I'd be okay with shipping base boxes for each version of PHP and adding some logic to pull down those version to start with it. I'm not too sure about others though because of the maintenance involved. Maybe we could add in additional configuration to point to different boxes so you could have say an "Altis Local Chassis" organisation in Vagrant Cloud then build a few base boxes to suit your needs and you could build and upload those to the cloud? That's my initial thoughts (pre coffee too) 😂. I'm definitely keen to hear @rmccue thoughts as he's always good with logic and options for new features.

**Comment by roborourke** [comment]
Oh yeah I'm not suggesting this project support any base boxes besides the latest one we roll out. This is purely for the secondary use case where an org can pre build their own and host them on Vagrant Cloud or wherever.

It does mean supporting the _ability_ to do that however but it's not a huge step away from the current steps we have in place.

**Comment by BronsonQuick** [comment]
@roborourke We've had a massive "rain event" in my city so I've been inside for too long and I had some free time for Chassis. I've added a way for people to specify their own boxes now. I've added some docs for it at the end of [this section](https://docs.chassis.io/en/latest/config/#base-box-mode).

Hopefully it should be pretty straight forward. I'm not sure if I should add some documentation around how people go about building and uploading their own boxes to Vagrant Cloud. I don't think too many people will use it so I think it's kinda outside of the scope of Chassis docs but I'm happy to something in if you think it's worth it.
